Docs: update s3 docs localstack instructions

REDDJF opened the pull request:
On the newest version of localstack, all APIs are exposed via a single edge service, which is accessible on port 4566.

geoand commented:
Thanks for this!

@marcinczeczko can you confirm this is needed?

REDDJF commented:
Here is a [related announcement](https://github.com/localstack/localstack/issues/2983) on the issue, which is also documented on the [project readme](https://github.com/localstack/localstack#overview)

geoand commented:
Cool.

Then maybe we need to some text in documentation mentioning that the **new** port is `4566` and linking to that issue?

REDDJF commented:
Something along the line of:
"**Note**: Versions of localstack newer than v0.11.5 require port `4566` instead of port `4572`. See this [GitHub issue](https://github.com/localstack/localstack/issues/2983) for details on this change"
?

geoand commented:
Yeah, that sounds good

geoand commented on docs/src/main/asciidoc/amazon-s3.adoc:
```diff
@@ -39,9 +39,9 @@ The easiest way to start working with S3 is to run a local instance as a contain
+docker run -it --publish 8008:4566 -e SERVICES=s3 -e START_WEB=0 localstack/localstack
 ----
-This starts a S3 instance that is accessible on port `8008`.
+This starts a S3 instance that is accessible on port `8008`. **Note**: Versions of localstack newer than v0.11.5 require port `4566` instead of port `4572`. See this https://github.com/localstack/localstack/issues/2983[GitHub issue] for details on this change
```
Minor nitpick: Can we have this on a separate line like so:

```
NOTE: Versions of localstack newer than v0.11.5....
```

This will make the rendering of the adoc as HTML much better.

Thanks!

geoand commented:
Wonderful!

A final ask if I may - can you please squash the commits into a single one? Because we use a merge-commit, GitHub doesn't allows us to automatically merge them from the UI

geoand reviewed (approved):
Thanks a lot!

REDDJF committed:
docs: update s3 docs localstack instructions

On the newest version of localstack, all APIs are exposed via a single
edge service, which is accessible on port 4566.
